File: tests/fixtures/cpp/boolean_operators.cpp
// Expected metrics:
// checkAll: cognitive=TBD, cyclomatic=TBD, sloc=TBD
bool checkAll(bool a, bool b, bool c) {
    if (a && b && c) {
        return true;
    }
    return false;
}

// checkMixed: cognitive=TBD, cyclomatic=TBD, sloc=TBD
bool checkMixed(bool a, bool b, bool c) {
    if (a && b || c) {
        return true;
    }
    return false;
}

File: tests/fixtures/rust/boolean_operators.rs
// Expected metrics:
// check_same: cognitive=2, cyclomatic=4, sloc=6
// check_mixed: cognitive=3, cyclomatic=4, sloc=6
fn check_same(a: bool, b: bool, c: bool) -> bool {
    if a && b && c {
        return true;
    }
    false
}

fn check_mixed(a: bool, b: bool, c: bool) -> bool {
    if a && b || c {
        return true;
    }
    false
}
